src/client/common/experiments/service.ts

```diff
@@ -57,6 +58,16 @@ export class ExperimentService implements IExperimentService {
         this._optInto = optInto.filter((exp) => !exp.endsWith('control'));
         this._optOutFrom = optOutFrom.filter((exp) => !exp.endsWith('control'));
 
+        // Custom settings just for native notebook support.
+        const settings = workspaceService.getConfiguration('workbench', undefined);
+        const editorAssociations = settings.get('editorAssociations') as {
+            viewType: string;
+            filenamePattern: string;
+        }[];
+        if (editorAssociations.find((a) => a.viewType && a.viewType.includes('jupyter-notebook'))) {
```

Comment:
Ewww, all of these hacks make me go 🤮 
But i don't see any other way around it, not complaining about your code, just the hacks we have for native notebooks & experiments (opting in & out...) too complicated

Reply:
I agree. I hate the experiments stuff too.